cli/cli-meta/src/index.ts

```diff
@@ -43,3 +43,18 @@ export function detectIfCurrentPkgIsExecutable (proc: NodeJS.Process = process):
 export function isExecutedByCorepack (env: NodeJS.ProcessEnv = process.env): boolean {
   return env.COREPACK_ROOT != null
 }
+
+export function getCurrentPackageName (): string {
+  return detectIfCurrentPkgIsExecutable() ? getExePackageName() : 'pnpm'
+}
+
+function getExePackageName (): string {
+  const platform = process.platform === 'win32'
+    ? 'win'
+    : process.platform === 'darwin'
+      ? 'macos'
+      : process.platform
+  const arch = platform === 'win' && process.arch === 'ia32' ? 'x86' : process.arch
```

Comment:
A less entangled code is easier to read and easier to change.

```suggestion
  const arch = process.platform === 'win32' && process.arch === 'ia32' ? 'x86' : process.arch
```

Reply:
I am not sure what is the suggested improvement, the platform variable is still used below. Why is using process.platform better?

Reply:
With `process.platform`, `arch`'s dependence on `platform` will be removed. All changes made to the `platform` expression would not affect `arch`. The code is less entangled, less chaotic, more predictable.